src/lib/audio/audio-effects.js

```diff
@@ -0,0 +1,81 @@
+import EchoEffect from './effects/echo-effect.js';
+import RobotEffect from './effects/robot-effect.js';
+import VolumeEffect from './effects/volume-effect.js';
+
+const effectTypes = {
+    ROBOT: 'robot',
+    REVERSE: 'reverse',
+    LOUDER: 'higher',
+    SOFTER: 'lower',
+    FASTER: 'faster',
+    SLOWER: 'slower',
+    ECHO: 'echo'
+};
+
+class AudioEffects {
+    static get effectTypes () {
+        return effectTypes;
+    }
+    constructor (buffer, name) {
+        // Some effects will modify the playback rate and/or number of samples.
+        // Need to precompute those values to create the offline audio context.
+        let sampleCount = buffer.length;
+        let playbackRate = 1;
+        switch (name) {
+        case effectTypes.ECHO:
+            sampleCount = buffer.length + 0.25 * 3 * buffer.sampleRate;
+            break;
+        case effectTypes.FASTER:
+            playbackRate = 1.5;
```

Comment:
The values for playback rate and volume change are not the same as the ones we playtested with. They should be updated to match.